src/core/meshcop/commissioner.cpp

```diff
@@ -636,6 +636,8 @@ void Commissioner::HandleLeaderPetitionResponse(Coap::Header *aHeader, Message *
     mTransmitAttempts = 0;
     mTimer.Start(Timer::SecToMsec(kKeepAliveTimeout) / 2);
 
+    SendCommissionerSet();
```

Comment:
I, personally, see both ways (checking for valid joiner entries and not checking) as viable solutions with different negligible trade-offs. First one removes one packet from the network, the other makes sure that the joiner set is always defined and saves irrelevant amount of codesize. If you prefer the first variant, I will add the check for valid joiner entries.